crates/solver/src/liquidity/zeroex.rs

```diff
@@ -79,36 +120,136 @@ impl ZeroExLiquidity {
     }
 }
 
+/// Calls the 0x API to get all orders and initializes the cache with them.
+async fn init_cache(
+    api: Arc<dyn ZeroExApi>,
+    cache: Arc<Mutex<HashMap<String, OrderRecord>>>,
+    sender: Address,
+) -> Result<()> {
+    cache.lock().await.clear(); // init can be called multiple times, so clear the cache first
```

Comment:
You are not holding the lock here. You could either not use a `tokio::Mutex` if you don't need to hold the lock across `.await`s or do:
```rust
let mut cache = cache.lock().await;
cache.clear();
```

Reply:
Indeed. Put it into a block.